contracts/PangolinExchange.sol

```diff
@@ -73,6 +89,36 @@ contract PangolinExchange is Ownable {
 
   /* ========== VIEW FUNCTIONS ========== */
 
+  /**
+    * Returns the current balance of the asset held by a given user
+    * @dev _asset the code of an asset
+    * @dev _user the address of queried user
+  **/
+  function getBalance(address _user, bytes32 _asset) external override view returns(uint256) {
+    IERC20 token = IERC20(getAssetAddress(_asset));
+    return token.balanceOf(_user);
+  }
+
+
+  /**
+     * Returns number of decimal places of a chosen asset given that it was previously set. Raises an error otherwise.
+     * @dev _asset the code of an asset
+  **/
+  function getAssetDecimalPlaces(bytes32 asset) external override view returns(uint256) {
```

Comment:
Please take a look at the first comment regarding reading Decimals from an ERC20 token contract

Reply:
I already answered there. Based on our discussion outcome I'll implement it using the IERC20Detailed interface.